src/index.js

```diff
@@ -1,124 +1,134 @@
 import Plugin from '@swup/plugin';
-import pathToRegexp from 'path-to-regexp';
+import { matchPath } from 'swup';
 
 export default class SwupJsPlugin extends Plugin {
 	name = 'SwupJsPlugin';
 
-	currentAnimation = null;
 	requires = { swup: '>=4' };
 
-	constructor(options = {}) {
-		super();
-		const defaultOptions = [
+	defaults = {
+		animations: [
 			{
 				from: '(.*)',
 				to: '(.*)',
 				out: (next) => next(),
 				in: (next) => next()
 			}
-		];
+		],
+		matchOptions: {}
```

Comment:
What is the intended use case for providing `matchOptions` as  an option that will apply to all `animations`? 

For the fragment plugin, I decided to not provide access to them at all, or if I do ever, I will likely provide them for every rule individually:

```js
{
	from: '(.*)',
	to: '(.*)',
	out: (next) => next(),
	in: (next) => next(),
	matchOptions: {}
}
```
My reasoning is that I would likely only want to alter the options for rules separately. For example, match only the tail of the current URL. Pretty much everything is already possible using wildcards, though: `(.*)/users/:user?`

Reply:
Having differing match options across a list of URLs sounds very confusing to me. I won't know how a URL will actually match unless I keep the differing options in my head at the same time. Do you have an example for where you'd need tail matching that doesn't work with wildcards?

Reply:
To be honest, I've never used the matchOptions in any of the plugins that offer them, so this might be a case of not having that option at all, and only adding it once someone requests it with a valid use case. But maybe you have one :)

Reply:
I don't have one, let's strip it :)